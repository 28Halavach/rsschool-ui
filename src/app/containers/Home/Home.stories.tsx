import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { ReduxProviderDecorator } from '../../storybook/ReduxProviderDecorator';
import Home from './index';

const noop = () => {};

storiesOf('Home', module)
    .addDecorator(storyFn => <ReduxProviderDecorator>{storyFn()}</ReduxProviderDecorator>)
    .add('No Data', () => {
        return <Home feed={[]} fetchFeed={noop} />;
    });
