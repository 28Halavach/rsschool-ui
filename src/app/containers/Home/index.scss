.Home {
  text-align: center;
}

.Home-header {
  height: 150px;
  padding: 20px;
}

.Home-title {
  // font-size: 1.5em;
}
