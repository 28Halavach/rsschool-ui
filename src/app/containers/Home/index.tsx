import { classNames } from 'core/styles';
import * as React from 'react';

const cn = classNames(require('./index.scss'));

const Home = (): JSX.Element => {
    return (
        <div className={cn('home-container')}>
            <header className={cn('home-header')}>
                <h1>Welcome to RS School</h1>
            </header>
        </div>
    );
};

export default Home;
