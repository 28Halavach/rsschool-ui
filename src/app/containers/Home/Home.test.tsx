import { shallow } from 'enzyme';
import * as React from 'react';
import Home from './index';
const { __stateGetter } = require('react-redux');

const noop = () => {};

describe('Home', () => {
    beforeEach(() => {
        __stateGetter.mockImplementation(() => ({}));
    });

    it('renders correctly if no data', () => {
        const output = shallow(<Home courses={[]} participations={[]} fetchFeed={noop} feed={[]} />);
        expect(output).toMatchSnapshot();
    });
});
