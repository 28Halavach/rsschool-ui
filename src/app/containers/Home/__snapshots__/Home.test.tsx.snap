// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly if no data 1`] = `
<Home
  courses={Array []}
  feed={Array []}
  fetchFeed={[Function]}
  participations={Array []}
/>
`;
